src/libraries/Microsoft.Extensions.Configuration.Binder/src/ConfigurationBinder.cs

```diff
@@ -532,10 +532,19 @@ private static bool CanBindToTheseConstructorParameters(ParameterInfo[] construc
             Type keyType = dictionaryType.GenericTypeArguments[0];
             Type valueType = dictionaryType.GenericTypeArguments[1];
             bool keyTypeIsEnum = keyType.IsEnum;
-
-            if (keyType != typeof(string) && !keyTypeIsEnum)
-            {
-                // We only support string and enum keys
+            bool keyTypeIsIntegralNonPtr =
+                keyType == typeof(sbyte) ||
+                keyType == typeof(byte) ||
+                keyType == typeof(short) ||
+                keyType == typeof(ushort) ||
+                keyType == typeof(int) ||
+                keyType == typeof(uint) ||
+                keyType == typeof(long) ||
+                keyType == typeof(ulong);
+
+            if (keyType != typeof(string) && !keyTypeIsEnum && !keyTypeIsIntegralNonPtr)
+            {
+                // We only support string, enum and integral (except nint-IntPtr and nuint-UIntPtr) keys
```

Comment:
```suggestion
                // We only support string, enum and integer (except nint-IntPtr and nuint-UIntPtr) keys
```

https://en.wikipedia.org/wiki/Integral
https://en.wikipedia.org/wiki/Integer

Reply:
Sorry for the delay: some holidays... I used the term "integral" trying to be more consistent witth C# language specs. wording... see [here](https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/builtin-types/integral-numeric-types) and [here](https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/language-specification/types#836-integral-types). I have also a long background with C++ where they also use the wording "integral types" to define such kind of types...
But i've to agree with you that "integer" will be more readable. I'm going to fix it ASAP